File: src/main/java/FindSmallestLetterGreaterThanTarget/zhangwentao/FindSmallestLetterGreaterThanTarget.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao

class Solution(object):
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """
        lettersOfSet = set(letters)
        result = None
        for letter in lettersOfSet :
            if result != None and target < letter and result > letter:
                result = letter
            elif result == None and target < letter :
                result = letter
        if result == None : result = letters[0]
        return result

File: src/main/go/FindSmallestLetterGreaterThanTarget/findSmallestLetterGreaterThanTarget.go
func nextGreatestLetter(letters []byte, target byte) byte {
    var min byte = target
    var max byte = target
    for _, letter := range letters {
        if letter < min {
            min = letter
        } else if letter > target {
            if max == target {
                max = letter
            } else if max > letter {
                max = letter
            }
        }
    }
    if max == target {
        return min
    } else {
        return max
    }
}
